src/record/api.rs

```diff
@@ -55,7 +56,59 @@ pub enum Row {
   Map(Vec<(Row, Row)>) // List of key-value pairs
 }
 
+// Macro to generate type-safe get_xxx methods e.g. get_bool, get_short
+macro_rules! row_primitive_accessor {
+  ($METHOD:ident, $VARIANT:ident, $TY:ty) => {
+    pub fn $METHOD(&self) -> Result<$TY, ParquetError> {
+        match *self {
+            Row::$VARIANT(v) => Ok(v),
+            _ => Err(ParquetError::General(format!("Cannot access {} as {}", self.get_type_name(), stringify!($VARIANT))))
```

Comment:
I was thinking if it would be easier to just print `self` instead of `self.get_type_name`? One of the pros is that we need not maintain the mapping to string values, but I am happy either way.

Reply:
If we print `self`, we would have a message like `Cannot access 1.2 as Bool` which makes debugging harder?

Reply:
Fair enough, let's keep type names!